src/frequenz/client/base/auth_interceptor.py

```diff
@@ -0,0 +1,77 @@
+# License: MIT
+# Copyright © 2025 Frequenz Energy-as-a-Service GmbH
+
+"""An Interceptor that adds the API key to a gRPC call."""
+
+import dataclasses
+from typing import Callable
+
+from grpc.aio import (
+    ClientCallDetails,
+    Metadata,
+    UnaryUnaryCall,
+    UnaryUnaryClientInterceptor,
+)
+
+
+@dataclasses.dataclass(frozen=True)
+class AuthenticationOptions:
+    """Options for authenticating to the endpoint."""
+
+    api_key: str
+    """The API key to authenticate with."""
+
+
+# There is an issue in gRPC that causes the type to be unspecifieable correctly here.
+class AuthenticationInterceptor(UnaryUnaryClientInterceptor):  # type: ignore[type-arg]
+    """An Interceptor that adds HMAC authentication of the metadata fields to a gRPC call."""
+
+    def __init__(self, *, authentication_options: AuthenticationOptions):
```

Comment:
Also nitpick/optional: Now that there is only one arg, and it is not a `str` (so arguments can't be mixed up) we could remove the keyword-only requirement, or if we keep it maybe use just `options` as argument name, since one can easily see which type of options they are from the type.

(same for the signing interceptor)